src/Symfony/Component/Console/Command/Command.php

```diff
@@ -159,10 +147,6 @@ public function __construct(?string $name = null, ?callable $code = null)
             $this->addUsage($usage);
         }
 
-        if (\is_callable($this) && self::class === (new \ReflectionMethod($this, 'execute'))->getDeclaringClass()->name) {
-            $this->code = new InvokableCommand($this, $this(...));
-        }
```

Comment:
This lines are here to support invokable classes that extend the `Command` class. This is a supported feature, see `InvokableCommandTest::testCallInvokeMethodWhenExtendingCommandClass` that is now failing.

I would add `if (!$code && ...` to the condition.